### PR TITLE
Bug 1537599 - Push when running apb test.

### PR DESCRIPTION
**Pull request by cfchase**

### CODE REVIEW

**Review by djzager — COMMENTED** [review]
LGTM. Finishing up something else and I'll test and approve. What's a good example apb to use?

**Comment by cfchase** [comment]
@djzager this is what I was using.  feel free to play with it.  https://github.com/cfchase/test-apb

File: src/apb/engine.py [hunk 1]
```diff
@@ -1232,6 +1232,9 @@ def cmdrun_test(**kwargs):
         kwargs['tag']
     )
 
+    registry = get_registry(kwargs)
```
Comment (djzager) [inline src/apb/engine.py]: If you have a look at the `cmdrun_run` you'll see we set the tag based on the registry like:

```
    apb_project = kwargs['base_path']
    registry = get_registry(kwargs)
    spec = get_spec(apb_project)
    tag = registry + "/" + kwargs['namespace'] + "/" + spec['name']

    image = build_apb(
        apb_project,
        kwargs['dockerfile'],
        tag
    )
    push_apb(registry, tag)
```

I think you will need to do something like that in order for this to work correctly.

**Review by djzager — CHANGES_REQUESTED** [review]
Just need to set the tag based on the registry before you build it in order for it to work correctly. After that I believe this will be ready.

File: src/apb/cli.py [hunk 2]
```diff
@@ -402,6 +402,38 @@ def subcmd_test_parser(subcmd):
         dest='dockerfile',
```
Comment (djzager) [inline src/apb/cli.py:395]: While you are at it, if you could remove the `tag` argument please.

**Review by djzager — APPROVED** [review]
ACK

**Review by dymurray — APPROVED** [review]

**Review by eriknelson — CHANGES_REQUESTED** [review]
I don't think this takes minishift  into account, but I'm not positive since you might be using the same functions a regular push does. Can you double check?

**Comment by dymurray** [comment]
@eriknelson I think it does actually. The minishift logic is handled in this function: https://github.com/ansibleplaybookbundle/ansible-playbook-bundle/blob/a2a180a85f7e387362214a65097efa4550724a77/src/apb/engine.py#L1338

`build_apb` and `push_apb` both call this function when building and pushing. @djzager am I right in that?

**Review by eriknelson — APPROVED** [review]

**Comment by eriknelson** [comment]
Confirmed it does use the shared logic. Thanks @dymurray, the other one I was concerned about was the tag, but that's also hidden behind `get_registry`. It's important that the internal minishift registry ip gets used when tagging images, if you're pushing to minishift.

**Pull request merged**